src/bin/main.rs

```diff
@@ -11,21 +9,9 @@ async fn main() -> anyhow::Result<()> {
     }
 
     env_logger::init();
+    let listener =
+        TcpListener::bind("0.0.0.0:7878").expect("Unable to bind the service in 0.0.0.0:7878");
```

Comment:
Yes, It if fine for the integration tests, we can use TcpListener to use random ports to spawn the servers in the tests

Reply:
I don't think this is the way to solve that. If we were intercepting this TCP stream someway by piping the stream for instance, creating our own TCP connection would make sense.

The issue here is strongly related to configuration and tests. What you want to avoid is having conflicts with OS sockets between the server and your tests suite.

This approach is affecting the way the server is implemented due to a rare case where the developer wants to run test while running the server.

You can easily fix this by relying on configuration and leave the server startup simple, clean and using just the minimal resources required.

I suggest defining environment variables with default values for this.

You would end up having predefined ports por both cases, server and tests, and you will be able to define them in the same place as all of the other configurations

Reply:
The problem is that I want a way to spawn servers in the integration tests with random ports, if I just decide to define a PORT from the environment for the tests, then we could have the "Address already in use" due to different tests are using the same port.